src/main/java/com/solace/spring/cloud/stream/binder/inbound/queue/FlowXMLMessageListener.java

```diff
@@ -108,13 +107,24 @@ public void stopReceiverThreads() {
     }
 
     @SuppressWarnings("BusyWait")
-    private void watchdog(long maxProcessingTimeMs) {
+    private void watchdog(long maxProcessingTimeMs, int threadCount) {
         while (running) {
             try {
                 if (solaceMeterAccessor.get() != null && bindingName.get() != null) {
                     solaceMeterAccessor.get().recordQueueSize(this.bindingName.get(), messageQueue.size());
                     solaceMeterAccessor.get().recordActiveMessages(this.bindingName.get(), activeMessages.size());
                 }
+
+                LocalDateTime now = LocalDateTime.now();
```

Comment:
Dont use LocalDateTime, this leads to unreliable and long running test.

Better wrap the time provider like in this example:

https://code.sbb.ch/projects/TP_TMS_PLATTFORM/repos/selfservice-portal-backend-solace/browse/src/main/java/ch/sbb/tms/ssp/utils/clock/Clock.java
https://code.sbb.ch/projects/TP_TMS_PLATTFORM/repos/selfservice-portal-backend-solace/browse/src/main/java/ch/sbb/tms/ssp/config/spring/ClockConfig.java

https://code.sbb.ch/projects/TP_TMS_PLATTFORM/repos/selfservice-portal-backend-solace/browse/src/test/java/ch/sbb/tms/ssp/utils/clock/MockClock.java
https://code.sbb.ch/projects/TP_TMS_PLATTFORM/repos/selfservice-portal-backend-solace/browse/src/test/java/ch/sbb/tms/ssp/service/solacebroker/SolaceCloudBrokerAutoUpdateServiceTestIT.java#125-130,266